models/custom.go

```diff
@@ -69,3 +75,142 @@ func (m *DomainResolutionRequestDomain) UnmarshalBinary(b []byte) error {
 	*m = res
 	return nil
 }
+
+// Value implements driver.Valuer for IP to convert Go string to PostgreSQL inet
+func (m IP) Value() (driver.Value, error) {
+	if m == "" {
+		return nil, nil
+	}
+
+	// Create a pgtype.Inet and set the value - it knows how to encode for PostgreSQL
+	var inet pgtype.Inet
+	if err := inet.Set(string(m)); err != nil {
+		return nil, fmt.Errorf("invalid IP address %q: %w", m, err)
+	}
+
+	// Return the pgtype.Inet's driver.Value
+	return inet.Value()
+}
+
+// Scan implements sql.Scanner for IP to convert PostgreSQL inet to Go string
+func (m *IP) Scan(value interface{}) error {
+	if value == nil {
+		*m = ""
+		return nil
+	}
+	switch v := value.(type) {
+	case string:
+		*m = IP(v)
+	case []byte:
+		*m = IP(string(v))
+	case *net.IPNet:
+		// pgx may return *net.IPNet for inet columns
+		if v != nil {
+			*m = IP(v.IP.String())
+		} else {
+			*m = ""
+		}
+	case net.IP:
+		*m = IP(v.String())
+	case *pgtype.Inet:
+		// pgtype.Inet is used by pgx for inet/cidr columns
+		if v != nil && v.Status == pgtype.Present && v.IPNet != nil {
+			*m = IP(v.IPNet.IP.String())
+		} else {
+			*m = ""
+		}
+	case pgtype.Inet:
+		if v.Status == pgtype.Present && v.IPNet != nil {
+			*m = IP(v.IPNet.IP.String())
+		} else {
+			*m = ""
+		}
```

Comment:
Aren't all IP columns mapped to `pgtype.Inet`? AFAIU this is the only case we'd need, wouldn't we?